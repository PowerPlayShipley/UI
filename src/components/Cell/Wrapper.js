import styled from "styled-components";

const Wrapper = styled.span`
  display: block;
`

export default Wrapper
