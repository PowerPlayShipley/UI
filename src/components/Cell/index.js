import React, { useState, useEffect } from 'react'
import PropTypes from 'prop-types'

import Wrapper from "./Wrapper";

// Going to keep these so we can add them to the span bellow just for prosperity
const Cell = ({ value: initialValue, row: { index }, column: { id }}) => {
  // We need to keep and update the state of the cell normally
  const [value, setValue] = useState(initialValue)

  // If the initialValue is changed external, sync it up with our state
  useEffect(() => {
    setValue(initialValue)
  }, [initialValue])

  return <Wrapper data-testid="cell-value" data-row={index} data-column={id}>{value}</Wrapper>
}

Cell.propTypes = {
  value: PropTypes.string.isRequired,
  row: PropTypes.shape({
    index: PropTypes.number
  }).isRequired,
  column: PropTypes.shape({
    id: PropTypes.number
  }).isRequired
}

export default Cell
