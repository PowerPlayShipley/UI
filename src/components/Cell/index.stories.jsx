import Cell from "./index";

export default {
  title: 'Cell/Cell',
  component: Cell
}

const Template = ({row, column, ...rest}) => {
  return (<Cell row={{ index: row }} column={{ id: column }} {...rest} />);
}

export const Default = Template.bind({});
Default.args = {
  row: 1,
  column: 1,
  value: ['A', 'B', 'C'].join(' ')
}
