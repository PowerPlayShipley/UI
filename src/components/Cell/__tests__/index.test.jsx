/**
 * @jest-environment jsdom
 */

import React from 'react'
import { render } from '@testing-library/react'

import Cell from "../index";

describe('Display Cell', () => {
  it('should render cell', () => {
    const { container } = render(<Cell column={{ id: 1 }} row={{ index: 1 }} value={'Hello'} />)
    expect(container).toHaveTextContent('Hello')
  })

  it('should create valid snapshot', function () {
    const { container } = render(<Cell column={{ id: 1 }} row={{ index: 1 }} value={'Hello'} />)
    expect(container.firstChild).toMatchSnapshot()
  });
})
