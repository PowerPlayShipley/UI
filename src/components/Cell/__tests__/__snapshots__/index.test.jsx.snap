// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display Cell should create valid snapshot 1`] = `
<span
  class=""
  data-column="1"
  data-row="1"
>
  Hello
</span>
`;
